src/common/bit_vector.hpp

```diff
@@ -224,6 +224,7 @@ class BitVector
   static void generate_test_instances(std::list<BitVector *> &o);
 private:
   struct NoInitAllocator : public std::allocator<__u32> {
+    NoInitAllocator() {}
```

Comment:
[OT]. `std::allocator::construct` is deprecated in C++17, and will be removed in C++20, probably we will have to do some nasty hacks. see  https://github.com/facebook/folly/blob/e40b0e4438592530b6772bf608f2f9a49251bdd3/folly/memory/UninitializedMemoryHacks.h#L268-L278